frontend/app/mocks/subscription-api.server.ts

```diff
@@ -0,0 +1,45 @@
+import { HttpResponse, http } from 'msw';
+
+import { db } from './db';
+import { getLogger } from '~/utils/logging.server';
+
+const log = getLogger('subscription-api.server');
+
+/**
+ * Server-side MSW mocks for the subscription API.
+ */
+export function getSubscriptionApiMockHandlers() {
+  log.info('Initializing Subscription API mock handlers');
+
+  return [
+    //
+    // Handler for GET request to retrieve email alerts decription by sin
+    //
+    http.get('https://api.example.com/dental-care/applicant-information/dts/v1/subscriptions', ({ request }) => {
+      log.debug('Handling request for [%s]', request.url);
+
+      const url = new URL(request.url);
+      const sin = url.searchParams.get('sin');
+      const alertType = url.searchParams.get('alertType');
+
+      if (sin === null || alertType === null) {
+        throw new HttpResponse(null, { status: 400 });
+      }
+
+      const subscriptionEntity = db.subscription.findFirst({
+        where: { sin: { equals: sin }, alertType: { equals: alertType } },
+      });
+
+      if (!subscriptionEntity) {
+        return HttpResponse.text(null, { status: 204 });
+      }
```

Comment:
When exposing an endpoint like `GET /subscriptions?sin={sin}`, the expected response should be a 200 with a collection of subscriptions for the `sin`.

Currently the mock API returns a single entity or 204. We should re-think how we get subscriptions by SIN as IT-Security won't like us sending a SIN query parameter... maybe send an ID instead (`GET /subscriptions/{id}`)

Reply:
removed